feat: Add input validation for gitURL and slug in POST /project endpoint

codeterrayt opened the pull request:
- Implemented validation to ensure gitURL is a valid GitHub URL
- Added validation to enforce proper format for the slug field, disallowing spaces
- Return appropriate error responses for invalid input formats

codeterrayt committed:
feat: Add input validation for gitURL and slug in POST /project endpoint

- Implemented validation to ensure gitURL is a valid GitHub URL
- Added validation to enforce proper format for the slug field, disallowing spaces
- Return appropriate error responses for invalid input formats

piyushgarg-dev commented on api-server/index.js:
```diff
@@ -37,7 +37,24 @@ app.use(express.json())
+        return res.status(400).json({ status:"error", error: 'Please Enter Valid Slug Format' });
+    }
+
+    const projectSlug = slug;
```
If there is no `slug` provided by the user, we should generate one.

codeterrayt replied:
So then I will have to remove the validation of slug from the request and generate the slug if it does not exist right?

codeterrayt committed:
fix: Generate slug if not present in /project endpoint

- Updated the /project endpoint to generate a new slug if not provided
- Resolves issue where endpoint previously returned an error response when slug was not present

codeterrayt committed:
Fixed Issue #7

Issue: Containers are still running #7
added: process.exit(0) to exit

codeterrayt commented:
hey @piyushgarg-dev Please have a look at changes
